apps/extension/src/entry/offscreen-handler.ts

```diff
@@ -13,13 +13,26 @@ chrome.runtime.onMessage.addListener((req, _sender, respond) => {
   if (isActionBuildRequest(request)) {
     void (async () => {
       try {
-        const data = await spawnActionBuildWorker(request);
+        // propagate errors that occur in unawaited promises
+        const unhandled = Promise.withResolvers<never>();
+        self.addEventListener('unhandledrejection', unhandled.reject, {
+          once: true,
+        });
+
+        const data = await Promise.race([
+          spawnActionBuildWorker(request),
+          unhandled.promise,
+        ]).finally(() => self.removeEventListener('unhandledrejection', unhandled.reject));
```

Comment:
Not sure I understand this. If you're already `await`ing `spawnActionBuildWorker`, won't the rejections throw an error that will then move use down into the `catch` block below? or are they "detached" somehow?

If the latter, this is a genius solution 🤯 

However, isn't there a risk that, while waiting for the action to build, an unrelated promise rejection could happen that this would catch, resulting in this action build getting killed?

Reply:
OK, thanks, this makes sense. FWIW, it's also possible that a completely unrelated promise rejection from some other JS process could trigger this failure too. But A) that seems like it'd be uncommon, and B) the worst case scenario is that the transaction fails to build, and must be re-submitted. That also seems acceptable.

Reply:
i believe unhandled promise rejections are document-scoped so this shouldn't be an issue